nucleus/core/bootstrap/src/main/java/com/sun/enterprise/glassfish/bootstrap/StaticGlassFishRuntime.java

```diff
@@ -171,84 +166,78 @@ private void setEnv(GlassFishProperties gfProps) throws Exception {
             new File(tmpDir).mkdirs();
         }
+
         File instanceRoot = File.createTempFile("gfembed", "tmp", new File(tmpDir));
-        if (!instanceRoot.delete() || !instanceRoot.mkdir()) { // convert the file into a directory.
+        // Convert the file into a directory.
+        if (!instanceRoot.delete() || !instanceRoot.mkdir()) {
             throw new Exception("cannot create directory: " + instanceRoot.getAbsolutePath());
         }
+
         try {
-            String[] configFiles = new String[]{"config/keyfile",
-                    "config/server.policy",
-                    "config/cacerts.jks",
-                    "config/keystore.jks",
-                    "config/login.conf",
-                    "config/admin-keyfile",
-                    "org/glassfish/web/embed/default-web.xml",
-                    "org/glassfish/embed/domain.xml"
+            String[] configFiles = new String[] {
+                "config/keyfile",
+                "config/server.policy",
+                "config/cacerts.jks",
+                "config/keystore.jks",
+                "config/login.conf",
+                "config/admin-keyfile",
+                "org/glassfish/web/embed/default-web.xml",
+                "org/glassfish/embed/domain.xml"
             };
-            /**
-             * Create instance config directory
-             */
+
+            // Create instance config directory.
             File instanceConfigDir = new File(instanceRoot, "config");
             instanceConfigDir.mkdirs();
-            /**
-             * Create instance docroot directory.
-             */
+
+            // Create instance docroot directory.
             new File(instanceRoot, "docroot").mkdirs();
-            /**
-             * Copy all the config files from uber jar to the instanceConfigDir
-             */
+
+            // Copy all the config files from uber jar to the instanceConfigDir.
             ClassLoader cl = getClass().getClassLoader();
             for (String configFile : configFiles) {
                 copy(cl.getResource(configFile), new File(instanceConfigDir,
-                        configFile.substring(configFile.lastIndexOf('/') + 1)), false);
+                    configFile.substring(configFile.lastIndexOf('/') + 1)), false);
             }
-            /**
-             * If the user has specified a custom domain.xml then copy it.
-             */
+
+            // If the user has specified a custom domain.xml then copy it.
             String configFileURI = gfProps.getConfigFileURI();
             if(configFileURI != null) {
-                copy(URI.create(configFileURI).toURL(),
-                        new File(instanceConfigDir, "domain.xml"), true);
+                copy(URI.create(configFileURI).toURL(), new File(instanceConfigDir, "domain.xml"), true);
             }
         } catch (Exception ex) {
-            ex.printStackTrace();
+            LOG.log(SEVERE, "Failed to create instanceRoot", ex);
```

Comment:
Can you do an experiment here for sure? If it is not possible to create an instance root -> it is not possible to load also logging.properties (maybe) -> no logging config available -> the error might end in the log buffer forever. Just try to throw an exception from this block and check if it is logged or not.
If not, revert this change and add comment why do we want to use STDERR instead of the JVM logging system.
I never know which part is executed  from which process. :-)

Reply:
@dmatej, it works ;)

![image](https://github.com/eclipse-ee4j/glassfish/assets/57197113/d1ba7262-8126-44e2-8456-7bcef4ae74fb)